File: day1/python/solution-part-two.py
import re


digit_mapping = {
    "zero": 0,
    "one": 1,
    "two": 2,
    "three": 3,
    "four": 4,
    "five": 5,
    "six": 6,
    "seven": 7,
    "eight": 8,
    "nine": 9,
    "0": 0,
    "1": 1,
    "2": 2,
    "3": 3,
    "4": 4,
    "5": 5,
    "6": 6,
    "7": 7,
    "8": 8,
    "9": 9,
}

with open("input.txt", "r") as file:

    data = sum(
        digit_mapping[
            re.search("(" + "|".join(digit_mapping.keys()) + ")", line).group(1)
        ]
        * 10
        + digit_mapping[
            re.search("(?s:.*)(" + "|".join(digit_mapping.keys()) + ")", line).group(1)
        ]
        for line in file
    )

print(data)

File: day1/javascript/solution-part-two.js
const { readFileSync } = require("fs");
const input = readFileSync("../input.txt", "utf-8");
const nums = {
    "one": 1,
    "two": 2,
    "three": 3,
    "four": 4,
    "five": 5,
    "six": 6,
    "seven": 7,
    "eight": 8,
    "nine": 9
};

const data = input.split('\n').map(line => {
    const digits = line.split('')
        .map((char, index) => {
            const remainingSubstring = line.slice(index);
            for (const [word, value] of Object.entries(nums)) {
                if (remainingSubstring.startsWith(word)) {
                    return value;
                }
            }
            return parseInt(char);
        })
        .filter(Boolean);

    return Number('' + digits[0] + digits[digits.length - 1]);
}).reduce((acc, c) => acc + c, 0);

console.log(data);
